js/api/Surveys.js

```diff
@@ -148,6 +169,21 @@ function clearSurveyCache(exclusions) {
     }
 
     realm.write(() => {
+      const expiredSurveysLength = expiredSurveys.length;
+      for (var i = 0; i < expiredSurveysLength; i++) {
+        let forms = realm.objects('Form').filtered(`surveyId= "${expiredSurveys[i].surveyId}"`)
+        let timeTriggers = realm.objects('TimeTrigger').filtered(`surveyId= "${expiredSurveys[i].surveyId}"`)
+        let geofenceTriggers = realm.objects('GeofenceTrigger').filtered(`surveyId= "${expiredSurveys[i].surveyId}"`)
+
+        let formsLength = forms.length;
+        for (var j = 0; j < formsLength; j++) {
+          let questions = realm.objects('Question').filtered(`formId= "${expiredSurveys[i].surveyId}"`)
```

Comment:
I'm already caching that expiredSurveys.length because of that problem with Realm debugging, forEach might have a similar effect to not having that cached perhaps?

I'll need to benchmark that at some point this weekend.